### PR TITLE
[oneTBB] Improve limiter_node decrementer spec

### PR DESCRIPTION
**Pull request by Iliamish**
Signed-off-by: ilya.mishin <ilya.mishin@intel.com>

### CODE REVIEW

File: source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst [hunk 1]
```diff
@@ -54,8 +54,12 @@ to the decrementer determine the value by which the internal counter of broadcas
 will be decreased, while negative values determine the value by which the internal
 counter of broadcasts will be increased.
 
+If as a template parameter ``DecrementType`` used ``continue_msg`` then the ``limiter_node``
+also acquires the behavior of the ``continue_node``.
+Internal threshold of the ``limiter_node`` decreases only when the decrementer receives
+the number of messages equal to the number of connected to decrementer predecessors.
 The ``continue_msg`` sent to the decrementer
-decreases the internal counter of broadcasts by one.
+decreases the internal counter of expected messages from predecessors by one.
```
Comment (aleksei-fedotov) [inline source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst:62]: Let's rewrite a bit:

If ``continue_msg`` is used as an argument for ``DecrementType`` template
parameter then the ``decrementer``'s port of the ``limiter_node`` also acquires the
behavior of the ``continue_node``, requiring the number of messages sent to it
be equal to the number of connected predecessors before it decreases the
internal counter of broadcasts by one.

**Review by aleksei-fedotov — CHANGES_REQUESTED** [review]

**Comment by aleksei-fedotov** [comment]
The technical part looks good to me. @alexandraepan, could you please review from the wording and stylistic standpoint.

File: source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst [hunk 2]
```diff
@@ -54,8 +54,11 @@ to the decrementer determine the value by which the internal counter of broadcas
 will be decreased, while negative values determine the value by which the internal
 counter of broadcasts will be increased.
 
-The ``continue_msg`` sent to the decrementer
-decreases the internal counter of broadcasts by one.
+If ``continue_msg`` is used as an argument for ``DecrementType`` template
+parameter then the ``decrementer``'s port of the ``limiter_node`` also acquires the
```
Comment (alexandraepan) [inline source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst:58]:
```suggestion
parameter, the ``decrementer``'s port of the ``limiter_node`` also acquires the
```

File: source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst [hunk 3]
```diff
@@ -54,8 +54,11 @@ to the decrementer determine the value by which the internal counter of broadcas
...
+behavior of the continue_node, requiring the number of messages sent to it
```
Comment (alexandraepan) [inline source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst:59]:
```suggestion
behavior of the ``continue_node``. This behavior requires the number of messages sent to it
```

File: source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst [hunk 4]
```diff
@@ -54,8 +54,11 @@ to the decrementer determine the value by which the internal counter of broadcas
...
+be equal to the number of connected predecessors before it decreases the
```
Comment (alexandraepan) [inline source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst:60]:
```suggestion
to be equal to the number of connected predecessors before decrementing the
```

File: source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst [hunk 5]
```diff
@@ -54,8 +54,11 @@ to the decrementer determine the value by which the internal counter of broadcas
 will be decreased, while negative values determine the value by which the internal
 counter of broadcasts will be increased.
 
-The ``continue_msg`` sent to the decrementer
-decreases the internal counter of broadcasts by one.
+If ``continue_msg`` is used as an argument for ``DecrementType`` template
```
Comment (alexandraepan) [inline source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst:57]:
```suggestion
If ``continue_msg`` is used as an argument for the ``DecrementType`` template
```

File: source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst (hunk 3 above)
Comment (alexandraepan) [reply source/elements/oneTBB/source/flow_graph/limiter_node_cls.rst:59]: Not sure what here **requires the number...** . If it's not the behavior but ``continue_node``, try just **It requires the...**

**Review by alexandraepan — APPROVED** [review]

**Comment by Iliamish** [comment]
@rscohn2 can you merge a request?

**Pull request merged**